doc/sphinxext/gen_rst.py

```diff
@@ -1,42 +1,83 @@
+#!/usr/bin/env python
+# -*- coding: utf-8
+
 """
 Example generation modified from the scikit learn
```

Comment:
-1 on the rest of these changes because they break the copy-paste / `diff` viewing with other repos